### PR TITLE
INT-B -19815- Office date validation/ alert - HHG, NTS, NTSR

### PR DESCRIPTION
**Pull request by MInthavongsay**
## Summary

**INTEGRATION**

This PR adds date validation for HHG, NTS and NTSR for requested pickup and/or delivery date selection for the shipment new/edit shipment views for SC and TOO roles.  It's a follow up to [PR-B -19813](https://github.com/transcom/mymove/pull/13426). An inline alert message will appear upon date selection from date picker. An alert message will appear on page load and on date selection. Message will be customized to respective date label. Alert will notify user if date selected is on weekend and/or holiday(federal holidays). For now implementation will assume date context is in the United States. For example, July 4th is a holiday and Sat/Sunday are considered weekend days.

Note: This PR contains code from [PR-B -19813](https://github.com/transcom/mymove/pull/13426)

## Verification Steps for the Author

These are to be checked by the author.

- [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
- [ ] Have the Agility acceptance criteria been met for this change?

## Verification Steps for Reviewers

These are to be checked by a reviewer.

- [ ] Has the branch been pulled in and checked out?
- [ ] Have the BL acceptance criteria been met for this change?
- [ ] Was the CircleCI build successful?
- [ ] Has the code been reviewed from a standards and best practices point of view?

### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

1. Login as customer and create/submit HHG, NTS and NTSR shipment types for a given move. For each select appropriate pickup/delivery dates that will test weekend/holiday alerts.
2. Login as service counselor and verify dates that are weekend and/or holiday for requested pickup/delivery dates display appropriate warning alerts on view load in EDIT view.  Verify alert messages are specific to either pickup or delivery. It must match corresponding date input label.
3. Verify changing dates to non weekend/holiday dates do not display warning alerts...etc...go through all test cases.
4. Verify alerts work when newly HHG, NTS, NTSR are added in service counselor.
5. Update orders with all required fields and submit MOVE DETAILS.
6. Login as TOO and access submitted in step 5. Verify alerts work  for all shipment type created/updated for move in SC role. Alerts should behave the same as in SC.
7. Again verify alerts work when newly HHG, NTS, NTSR are added in TOO.  
8. Verify DateIsWeekendHoliday API in Office swagger.
9. Try to break it and go through all possible use cases you can think of!

### Frontend

- [ ] There are no aXe warnings for UI.
- [ ] This works in [Supported Browsers and their phone views](https://transcom.github.io/mymove-docs/docs/adrs/Browser-Support/#minimum-browser-requirements) (Chrome, Firefox, Edge).
- [ ] There are no new console errors in the browser devtools.
- [ ] There are no new console errors in the test output.
- [ ] If this PR adds a new component to Storybook, it ensures the component is fully responsive, OR if it is intentionally not, a wrapping div using the `officeApp` class or custom `min-width` styling is used to hide any states the would not be visible to the user.
- [ ] This change meets the standards for [Section 508 compliance](https://www.ssa.gov/accessibility/andi/help/install.html).

### Backend

- [ ] Code follows the guidelines for [Logging](https://transcom.github.io/mymove-docs/docs/getting-started/development/logging).
- [ ] The requirements listed in [Querying the Database Safely](https://transcom.github.io/mymove-docs/docs/backend/guides/golang-guide#querying-the-database-safely) have been satisfied.

### CODE REVIEW

File: src/shared/calendar.js
Comment (antgmann) [inline src/shared/calendar.js:1]: This is the file that the bot wants moved, likely to src/components https://transcom.github.io/mymove-docs/docs/frontend/testing/frontend/#file-layout--naming

Comment (antgmann) [reply src/shared/calendar.js:1]: I don't know if that change is feasible though, just identifying.

Comment (ajlusk) [reply src/shared/calendar.js:1]: I was thinking `/src/utils` rather than `sr/components`

Comment (antgmann) [reply src/shared/calendar.js:1]: Agreed!

**Comment by ajlusk** [comment]
Changes generally look good so far but it looks like they may have broken a playwright test or two. Waiting to finish review until those are passing or determined to be flaky or unrelated.

File: src/shared/calendar.js
Comment (MInthavongsay) [reply src/shared/calendar.js:1]: I moved calendar.js to utils directory.

**Pull request closed without merging**